aws/solutions/DirectoryServiceSettings/src/directory_settings_custom_resource.py

```diff
@@ -0,0 +1,233 @@
+        return alias
+    else:
+        error_message = f"Directory already has a different alias.  Use '{existing_alias}' for the 'DirectoryAlias' CloudFormation parameter."
+        raise Exception(error_message)
+
+
+def enable_directory_sso(directory_id: str, existing_sso_enabled: bool):
+    """Enables single sign-on for the directory.
+
+    Args:
+        directory_id: Directory ID
+        existing_sso_enabled: Status of single sing-on for the directory
+    """
+    if not existing_sso_enabled:
+        response = ds_client.enable_sso(DirectoryId=directory_id)
+        logger.info("Directory enabled for SSO")
+        logger.debug(f"enable_sso_response = {json.dumps(response, default=str)}")
+    else:
+        logger.info("Directory was already enabled for SSO")
+
+
+def disable_directory_sso(directory_id: str, existing_sso_enabled: bool):
+    """Disables single sign-on for the directory.
+
+    Args:
+        directory_id: Directory ID
+        existing_sso_enabled: Status of single sing-on for the directory
+    """
+    if existing_sso_enabled:
+        response = ds_client.disable_sso(DirectoryId=directory_id)
+        logger.info("Directory disabled for SSO")
+        logger.debug(f"disable_sso_response = {json.dumps(response, default=str)}")
+    else:
+        logger.info("Directory was already disabled for SSO")
+
+
+@helper.create
+@helper.update
+def create_and_update(event, context):
+    """Create/Update Event from AWS CloudFormation.
+
+    Args:
+        event: event data
+        context: runtime information
+
+    Raises:
+        Exception: Captures all exceptions.
+    """
+    try:
+        logger.info(f"{event['RequestType']} Event")
+        logger.info(f"REQUEST RECEIVED: {json.dumps(event, default=str)}")
+        directory_id: str = event["ResourceProperties"]["DirectoryId"]
+        create_alias: str = event["ResourceProperties"]["CreateDirectoryAlias"]
+        enable_sso: str = event["ResourceProperties"]["EnableDirectorySSO"]
+        alias: str = event["ResourceProperties"]["DirectoryAlias"]
+        topic: str = event["ResourceProperties"]["DirectoryMonitoringTopicName"]
+        # Directory Monitoring
+        register_directory_monitoring_topic(directory_id, topic)
+        # Existing Alias & SSO Status
+        existing_alias, existing_sso_status = get_directory_alias_and_sso_enabled_status(directory_id)
+        logger.info(f"existing_alias={existing_alias} --- existing_sso_status={existing_sso_status}")
+        # Directory Alias
+        if create_alias == "Yes":
+            create_directory_alias(directory_id, alias, existing_alias)
+            helper.Data.update({"AliasUrl": f"https://{alias}.awsapps.com"})
+        # Directory SSO
+        if enable_sso == "Yes":
+            enable_directory_sso(directory_id, existing_sso_status)
+        else:
+            disable_directory_sso(directory_id, existing_sso_status)
+        helper.Data.update({"AliasUrl": ""})
+    except Exception as error:
```

Comment:
As per other PR, catching and immediately raising the exception can just be not catching it in the first place :)